architecture/src/commonMain/kotlin/observable/FlowSubject.kt

```diff
@@ -65,8 +65,18 @@ open class StateFlowInitializedSubject<T>(
         coroutineScope,
         context,
         { observedStateFlow },
-        { observedStateFlow.value = it },
-        { observedStateFlow.value = it },
+        {
+            if (observation.currentOrNull != it) {
+                observation.setValue(ObservableOptional.Value(it))
+            }
+            observedStateFlow.value = it
+        },
+        {
+            if (observation.observedValue.valueOrNull != it) {
```

Comment:
technically if this is `OO.Nothing` here, I think it should be replaced with the `Value` if that value is `null`